Ревью бэкенда

Feelcover opened the pull request.

kmvpw commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,25 @@
+name: Freelance platform workflow
+
+on: [push]
```
Любой пуш сопровождается запуском тестов. Так и планировали?

inigmat replied:
Изменил на запись только при push в develop, исключил запуск при изменения в общих файлах 

`on:
  push:
    paths-ignore:
      - 'README.md'
      - '.gitignore'
      - 'setup.cfg'
    branches:
      - develop
`

kmvpw commented on backend/api/admin.py:
```diff
@@ -0,0 +1,33 @@
+    list_filter = ('name', 'slug',)
+
+
+@admin.register(Member)
```
Почему решили разный подход использовать для регистрации класса?

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,193 @@
+                           StackJob)
+
+# File requiremnts
+MAX_FILE_SIZE_MB = 2
```
Мне кажется эти константы можно было перенести в файл с настройками. Иначе настройки приложения становятся "размазанными" по файлам с реализацией.

inigmat replied:
Константы из приложений api, chat, orders вынесены в settings

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,193 @@
+        user = data['user']['id']
+        job = data['job']['id']
+        if Response.objects.filter(user__id=user, job__id=job).exists():
+            raise serializers.ValidationError(
```
В начале файла в качестве констант вынесены сообщения об ошибках. В данном валидаторе используется сообщение жестко заданное в валидаторе. Чем это сообщение отличается от сообщений, для которых содержимое вынесено в отдельные константы?

inigmat replied:
Константы из приложений api, chat, orders вынесены в settings

kmvpw commented on backend/api/utils.py:
```diff
@@ -0,0 +1,31 @@
+from pdf2image import convert_from_bytes, convert_from_path
+from PIL import Image
+
+THUMBNAIL_SIZE = (100, 100)
```
Рекомендую выносить настройки в одно место в приложении.

inigmat replied:
Константы из приложений api, chat, orders вынесены в settings

kmvpw commented on backend/api/utils.py:
```diff
@@ -0,0 +1,31 @@
+        return thumbnail_buffer
+    except Exception as e:
+        print(f"Ошибка при создании миниатюры: {e}")
+        return None
```
Лишняя строка.

inigmat replied:
строку  return None удалил

kmvpw commented on backend/api/utils.py:
```diff
@@ -0,0 +1,31 @@
+    Создание миниатюр загруженных файлов в профайле заказа.
+    """
+    try:
+        if file.name.endswith(('.pdf', '.PDF')):
```
Не ясно - почему вилка в логике исходя из расширения файла?

inigmat replied:
Блок обработки pdf файлов уже удалил за неактуальностью

kmvpw commented on backend/api/views.py:
```diff
@@ -0,0 +1,64 @@
+# ...
+
+
+class JobViewSet(ModelViewSet):
```
Нужно описание (комментарий) - что это такое?

inigmat replied:
Описание добавлено

kmvpw reviewed (commented):
Для pull request - хорошая практика сопровождать pull request описанием. Например, какие бизнес задачи были решены.

Проблема с комментарии - где-то по коду есть, где-то вообще нет ни одного.

Постарайтесь разделять pull requests с изменениями бека и фронта

kmvpw commented on backend/api/management/commands/loaddata.py:
```diff
@@ -0,0 +1,27 @@
+                )
+                obj.save()
+
+        self.stdout.write(
```
Для чего реализовали вывод в консоль?

inigmat replied:
В консоль выводится сообщение об успешности загрузки данных - для контроля.

kmvpw commented on backend/api/permissions.py:
```diff
@@ -0,0 +1,119 @@
+                or request.user.is_authenticated)
+
+    def has_object_permission(self, request, _view, obj):
+        return (request.method in permissions.SAFE_METHODS
```
С объектом связана только одна проверка (последняя). Остальные должны выполняться в has_permission.

inigmat replied:
Удалили этот permission, так как уже не использовался.

kmvpw commented on backend/api/permissions.py:
```diff
@@ -0,0 +1,119 @@
+        return False
+
+
+''' Удалить если ничего не сломается
```
Хорошая практика - удалять код, который не используется (закомментирован). 
Через некоторое время не останется понимания, почему этот код оставили и сложно будет решить - можно ли его удалить или он действительно нужен.

inigmat replied:
Неиспользуемый код удалил.

kmvpw commented on backend/api/permissions.py:
```diff
@@ -0,0 +1,119 @@
+        return (request.method in permissions.SAFE_METHODS
+                or request.user.is_admin
+                or request.user.is_customer
+                or obj.client == request.user)
```
Остальные проверки нужно перенести в has_permission

inigmat replied:
Все проверки за исключением obj.client == request.user перенес в has_permission.

kmvpw commented on backend/api/permissions.py:
```diff
@@ -0,0 +1,119 @@
+                or obj.client == request.user)
+
+
+class ChatPermission(permissions.BasePermission):
```
Сложный класс, отлично, что решили оставить комментарий!

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,284 @@
+from users.clients import GetCustomerProfileSerializer
+from users.freelancers import GetWorkerProfileSerializer
+
+# File requiremnts
```
Хороший подход - используемые в коде параметры выносить в константы.

inigmat replied:
Спасибо за оценку и в целом за ревью!

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,284 @@
+                  'description', 'job_files',)
+
+    def validate_stack(self, data):
+        stack = self.initial_data.get('stack')
```
Почему здесь проверка не data, а информации из self.initial_data.get('stack') ?

inigmat replied:
Так как при попытке получения через словарь данных self.data - возникала ошибка 
```When a serializer is passed a `data` keyword argument you must call `.is_valid()` before attempting to access the serialized `.data` representation.
You should either call `.is_valid()` first, or access `.initial_data` instead.``` 

Убрал строчку  `stack = self.initial_data.get('stack')` - сделел сразу проверку   ` if data == []:`

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,284 @@
+        ask_budget = validated_data.pop('ask_budget', False)
+        ask_deadline = validated_data.pop('ask_deadline', False)
+        if ask_budget:
+            validated_data['budget'] = 'Жду предложений'
```
Почему такое значение? Судя по названию - должна быть сумма

inigmat replied:
По проекту у нас при создании задания заказчик вместо указания бюджета может запросить стоимость у фрилансера, - для этого в модели есть числовое поле budget и поле с булевым значением ask_budget. Поэтому если в ask_budget при создании заказа передано True - в БД в поле budget записывается 'Жду предложений', аналогично с deadline. Может как-то и по другому это можно решить - но пока не знаю как.

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,284 @@
+        if ask_budget:
+            validated_data['budget'] = 'Жду предложений'
+        if ask_deadline:
+            validated_data['deadline'] = 'Жду предложений'
```
Почему такое значение? Судя по названию - должна быть дата

inigmat replied:
По проекту у нас при создании задания заказчик вместо указания бюджета может запросить стоимость у фрилансера, - для этого в модели есть числовое поле budget и поле с булевым значением ask_budget. Поэтому если в ask_budget при создании заказа передано True - в БД в поле budget записывается 'Жду предложений', аналогично с deadline. Может как-то и по другому это можно решить - но пока не знаю как.

kmvpw commented on backend/api/serializers.py:
```diff
@@ -0,0 +1,284 @@
+                job=job, stack=stack_obj
+            )
+        job.category.set(category_data)
+        job.is_responded = False
```
Почему явно нужно задавать значение? Почему не задать значение в модели?

inigmat replied:
В модели поле отсутствует - оно добавлено с помощью в сериализаторе SerializerMethodField.

kmvpw reviewed (commented):
Описание PR - какие задачи были решены, на что обратить внимание

По тестам - не хватает минимального набора тестов реализованной логики. Сделайте проверки api, который передаете фронтам. Проверку, что API есть, с ним можно работать.

Хочу отметить - хорошо поработали над README.md 

Вижу комментарии от прошлого ревью, просьба закрвать PR и создавать новый для ревью. Иначе много кода накапливается, сложно проводить.

В ревью попало ревью фронта

inigmat committed:
добавил EMAIL_HOST_PASSWORD в workflow

inigmat committed:
Merge pull request #190 from freelancing-platform-practicum/refact/setemailhost

добавил EMAIL_HOST_PASSWORD в workflow

MaximParamonov committed:
убрал возможность автосоздания профиля

rakleed committed:
restore c0efc08 from @inigmat: добавил EMAIL_HOST_PASSWORD в workflow

rakleed committed:
restore d5d6e61 from @KalinaFrontend: feat: реализовать применение фильтров

rakleed committed:
fix: исправить авторизацию с невалидным access токеном

rakleed committed:
refactor: улучшить сообщения об ошибках при валидации

rakleed committed:
refactor: добавить проверку логина перед рендерингом сайта

rakleed committed:
fix: добавить очистку `session` и `localStorage` при выходе из аккаунта

rakleed committed:
fix: актуализировал стили на странице входа и регистрации

rakleed committed:
fix: добавить отступы для кнопки «показать пароль»

rakleed committed:
refactor: актуализировать валидацию в соответствии с фич-листом

MaximParamonov committed:
убрал вывод контактной информации для общего обозрения

MaximParamonov committed:
Merge pull request #191 from freelancing-platform-practicum/bugfix/users

Bugfix/users

MaximParamonov committed:
убрал вывод контактной информации для общего обозрения

MaximParamonov committed:
Merge pull request #192 from freelancing-platform-practicum/bugfix/users

убрал вывод контактной информации для общего обозрения

inigmat committed:
рефакторинг по результатам ревью

inigmat committed:
Merge branch 'develop' of https://github.com/freelancing-platform-practicum/freelancing-platform-project into develop

inigmat committed:
Merge branch 'develop' of https://github.com/freelancing-platform-practicum/freelancing-platform-project into refact/backend-after-review

inigmat committed:
Merge pull request #193 from freelancing-platform-practicum/refact/backend-after-review

Refact/backend after review

MaximParamonov committed:
Обновил валидаторы для профиля Фрилансер. Восстановил автосоздание профиля при необходимости

MaximParamonov committed:
Merge pull request #194 from freelancing-platform-practicum/bugfix/users

Обновил валидаторы для профиля Фрилансер.
categories - обязательно
user - вводить не обязательно. но если вводить, то при POST оба поля обязательно, при PATCH - только заменяемое поле
stacks - вводить не обязательно, но если вводить, то 'name' - обязательно
contacts - вводить не обязательно, но если вводить, то 'type' - обязательно
education- вводить не обязательно, но если вводить, то 'name', 'faculty' - обязательно

все остальные поля на усмотрение

Восстановил автосоздание профиля при необходимости

inigmat committed:
Merge branch 'develop' of https://github.com/freelancing-platform-practicum/freelancing-platform-project into develop

inigmat committed:
fix отображения списка заданий

inigmat committed:
Merge pull request #195 from freelancing-platform-practicum/fix/orderspermission

Fix/orderspermission

MaximParamonov committed:
реализовал вывод информации о фрилансерах, реализован поиск по ставке и специализации

inigmat committed:
добавил ограничение на колчичестов навыков в заказе

inigmat committed:
Merge pull request #197 from freelancing-platform-practicum/feat/back-job-stack-maxnumber

добавил ограничение на колчичестов навыков в заказе

MaximParamonov committed:
изменил поле по которому происходит сортировка, установил ловушки для swagger & redoc исключающие их ошибки

MaximParamonov committed:
выполнил слияние с develop и bugfix/users в которые внесли изменения

MaximParamonov committed:
выполнил слияние с develop и bugfix/users в которые внесли изменения

MaximParamonov committed:
исправил баг с лишним поле в данных заказчика

MaximParamonov committed:
готово к деплою

MaximParamonov committed:
готово к деплою

MaximParamonov committed:
поправил README.md

inigmat committed:
исправил подсчет количества тэгов в задании

inigmat committed:
Merge pull request #203 from freelancing-platform-practicum/fix/back-orders-stack

исправил подсчет количества тэгов в задании

inigmat committed:
добавил файлы подтверждения домена для почты

inigmat committed:
Merge pull request #204 from freelancing-platform-practicum/backend/email-sparkpost

добавил файлы подтверждения домена для почты

inigmat committed:
переместил папку zohoverify и добавил url

inigmat committed:
Merge pull request #205 from freelancing-platform-practicum/backend/email-sparkpost

переместил папку zohoverify и добавил url

inigmat committed:
обновил nginx для zohoverify

inigmat committed:
Merge pull request #206 from freelancing-platform-practicum/backend/email-sparkpost

обновил nginx для zohoverify

inigmat committed:
обновил nginx для zohoverify

inigmat committed:
Merge pull request #207 from freelancing-platform-practicum/backend/email-sparkpost

обновил nginx для zohoverify

inigmat committed:
обновил nginx и docker compose для zohoverify

inigmat committed:
Merge pull request #208 from freelancing-platform-practicum/backend/email-sparkpost

обновил nginx и docker compose для zohoverify

inigmat committed:
обновил nginx и docker compose для zohoverify

inigmat committed:
Merge pull request #209 from freelancing-platform-practicum/backend/email-sparkpost

обновил nginx и docker compose для zohoverify

inigmat committed:
обновил nginx и docker compose для zohoverify

inigmat committed:
Merge pull request #210 from freelancing-platform-practicum/backend/email-sparkpost

обновил nginx и docker compose для zohoverify

inigmat committed:
пока убрал в  nginx и docker compose для zohoverify

inigmat committed:
Merge pull request #211 from freelancing-platform-practicum/backend/email-sparkpost

пока убрал в  nginx и docker compose для zohoverify

inigmat committed:
перенес zohoverify в static

inigmat committed:
Merge pull request #212 from freelancing-platform-practicum/backend/email-sparkpost

перенес zohoverify в static

inigmat committed:
подправил dockerfile backend

inigmat committed:
Merge pull request #213 from freelancing-platform-practicum/backend/email-sparkpost

подправил dockerfile backend

inigmat committed:
подправил workflow

inigmat committed:
Merge pull request #214 from freelancing-platform-practicum/backend/email-sparkpost

подправил workflow

inigmat committed:
добавил копирование в  dockerfile backend

inigmat committed:
Merge pull request #215 from freelancing-platform-practicum/backend/email-sparkpost

добавил копирование в  dockerfile backend

inigmat committed:
переместил папку zohoverify

inigmat committed:
Merge pull request #216 from freelancing-platform-practicum/backend/email-sparkpost

переместил папку zohoverify

alexandraspage committed:
регистрация заказчика

rakleed committed:
refactor: добавить вывод информации с бэкенда о фрилансере и заказчике в профиле

Также исправил некоторые замечания дизайнеров

MaximParamonov committed:
устранил проблему с созданием второго профиля заказчика, исправил проблему с загрузкой фото фрилансера. только png

MaximParamonov committed:
Merge pull request #217 from freelancing-platform-practicum/bugfix/users

устранил проблему с созданием второго профиля заказчика, исправил про…

rakleed committed:
feat: открывать профиль после успешного создания профиля

rakleed committed:
refactor: добавить информацию о пользователе в контекст при успешном создании профиля

inigmat committed:
добавил client_max_body_size в конфиг nginx

inigmat committed:
Merge pull request #218 from freelancing-platform-practicum/fix/nginx-conf

добавил client_max_body_size в конфиг nginx

rakleed committed:
fix: исправить `TypeError` при открытии профиля фрилансера без прикреплённых файлов образования

rakleed committed:
fix: исправить отображение фото в хэдере

inigmat committed:
добавил client_max_body_size в конфиг nginx

inigmat committed:
Merge pull request #219 from freelancing-platform-practicum/fix/nginx-conf

добавил client_max_body_size в конфиг nginx

rakleed committed:
feat: отправка данных на бекенд при редактировании профиля

alexandraspage committed:
заполнение профиля фрилансера

alexandraspage committed:
фрилансер регистрация

MaximParamonov committed:
Добавил вывод имени и фамилии в профиле заказчика и контактов фрилансера

MaximParamonov committed:
Merge pull request #220 from freelancing-platform-practicum/bugfix/users

Добавил вывод имени и фамилии в профиле заказчика и контактов фрилансера

alexandraspage committed:
навыки

rakleed committed:
feat: отправка данных на бекенд при редактировании профиля фрилансера

inigmat committed:
fix создания заказа - слаг для указания спецаилизации и исправление ошибки передачи запросо сроков и бюджета

inigmat committed:
Merge pull request #221 from freelancing-platform-practicum/fix/backend_jobcreate

fix создания заказа - слаг для указания спецаилизации и исправление о…

inigmat committed:
fix ошибки указания сроков и бюджета

inigmat committed:
Merge pull request #222 from freelancing-platform-practicum/fix/backend_jobcreate

fix ошибки указания сроков и бюджета

inigmat committed:
fix подправил тексты ошибок - вынес в settings

inigmat committed:
Merge pull request #223 from freelancing-platform-practicum/fix/backend_jobcreate

fix подправил тексты ошибок - вынес в settings

inigmat committed:
fix фильтров заданий

inigmat committed:
Merge pull request #224 from freelancing-platform-practicum/fix/back-jobfilters

fix фильтров заданий

inigmat committed:
fix фильтров заданий

inigmat committed:
Merge pull request #225 from freelancing-platform-practicum/fix/back-jobfilters

fix фильтров заданий

inigmat committed:
fix фильтров заданий

inigmat committed:
Merge pull request #226 from freelancing-platform-practicum/fix/back-jobfilters

fix фильтров заданий

alexandraspage committed:
создать заказ

inigmat committed:
fix отображения запрос бюджета и формата даты

inigmat committed:
Merge pull request #227 from freelancing-platform-practicum/fix/backend-jobs

fix отображения запрос бюджета и формата даты

inigmat committed:
fix порядка отображения заданий

inigmat committed:
Merge pull request #228 from freelancing-platform-practicum/fix/backend-jobs

fix порядка отображения заданий

inigmat committed:
добавил id для сustomer

inigmat committed:
Merge pull request #229 from freelancing-platform-practicum/fix/back-customerid

добавил id для сustomer

inigmat committed:
добавил id для сustomer

inigmat committed:
Merge pull request #230 from freelancing-platform-practicum/fix/back-customerid

добавил id для сustomer

alexandraspage committed:
фильтры на главной

rakleed committed:
fix: сделать цвета в карточке заказа на главной как в макете

rakleed committed:
fix: убрать отступ между после названия в карточке заказа на главной

rakleed committed:
fix: увеличить длину описания таски тестировщика

alexandraspage committed:
фото профиля

inigmat committed:
переезд на новый сервер

inigmat committed:
переезд на новый сервер -  добавил host в settings

inigmat committed:
установил ветку develop для github actions

inigmat committed:
Merge pull request #231 from freelancing-platform-practicum/cicd/new-host

Cicd/new host

inigmat committed:
переименовал команду загрузки категорий

inigmat committed:
Merge pull request #232 from freelancing-platform-practicum/cicd/new-host

переименовал команду загрузки категорий

rakleed committed:
chore(front): обновить правила линтера и добавить форматер

rakleed committed:
fix(front): сделать корректный экспорт констант

rakleed committed:
fix(front): передавать корректную дату при создании заказа

rakleed committed:
chore(front): добавить и настроить правила линтера

alexandraspage committed:
загрузка файлов

alexandraspage committed:
get freelancers

MaximParamonov committed:
добавил вывод фото фрилансера на главной странице

MaximParamonov committed:
добавил вывод фото фрилансера на главной странице

MaximParamonov committed:
добавил вывод фото фрилансера на главной странице

MaximParamonov committed:
Merge pull request #234 from freelancing-platform-practicum/bugfix/users

Bugfix/users

alexandraspage committed:
ошибки

alexandraspage committed:
ошибки

rakleed committed:
feat(front): вывести на главной таски/фрилансеров с бэкенда

inigmat committed:
отображения для заданий данных клиента и запросо срока и бюджета

inigmat committed:
Merge pull request #246 from freelancing-platform-practicum/fix/back-job-askinfo

отображения для заданий данных клиента и запросо срока и бюджета

inigmat committed:
отображения для заданий данных клиента изменение

inigmat committed:
отображения для заданий данных клиента изменение

inigmat committed:
Merge pull request #247 from freelancing-platform-practicum/fix/back-job-askinfo

Fix/back job askinfo

sobbaka committed:
add: валидация полей имя и отрасль для заказчика

inigmat committed:
fix отображения данных заданий и отклика

inigmat committed:
Merge pull request #249 from freelancing-platform-practicum/fix/back-jobinfo

fix отображения данных заданий и отклика

sobbaka committed:
add: валидация О компании и Сайт заказчика

alexandraspage committed:
Merge pull request #250 from freelancing-platform-practicum/customer_validation

Валидация анкеты заказчика (регистрация)

inigmat committed:
добавил эндпоинт со списком откликов на задание

inigmat committed:
Merge pull request #251 from freelancing-platform-practicum/feat/back-list-freelance-responses

добавил эндпоинт со списком откликов на задание

inigmat committed:
upgrade эндпоинта добавил фильтр и поиск

inigmat committed:
Merge pull request #252 from freelancing-platform-practicum/feat/back-list-freelance-responses

upgrade эндпоинта добавил фильтр и поиск

sobbaka committed:
ref: изменил название handleEmployeChange

sobbaka committed:
add: добавил пробелы в nameRegex и aboutRegex

sobbaka committed:
add: валидация формы заказчика ProfileCustomer

rakleed committed:
feat(front): отображение информации о заказе

inigmat committed:
исправил сохрание значений переменных ask budget deadline

inigmat committed:
Merge pull request #253 from freelancing-platform-practicum/fix/back-job-askfields

fix не выполнялась запись в БД значений переменных ask_budget и ask_deadline

rakleed committed:
fix(front): вернуть ссылки на открытие информации о заказе/фрилансере

MaximParamonov committed:
правки по просьбе фронтэда

rakleed committed:
fix(front): корректное получение данных о задачах для фрилансера на главной

rakleed committed:
feat(front): отображение информации о фрилансере с бэкенда

- Улучшения отображения информации о заказе
- Единый URL для профиля у фрилансера и заказчика

rakleed committed:
fix(front): исправить проблему с некорректной валидацией повторного пароля

rakleed committed:
Revert "fix(front): исправить проблему с некорректной валидацией повторного пароля"

This reverts commit 1ebea8db8010474826008e36b45ad63d983fca46.

sobbaka committed:
add: валидация формы заказчика

sobbaka committed:
ref: поправил isDisabled

rakleed committed:
fix(front): показывать информацию о заказе для любого фрилансера

sobbaka committed:
ref: вынес отдельно initialData

MaximParamonov committed:
правки по просьбе фронтэда

MaximParamonov committed:
Merge branch 'develop' into bugfix/users

MaximParamonov committed:
Merge pull request #262 from freelancing-platform-practicum/bugfix/users

Bugfix/users

sobbaka committed:
fix: форма невалидна по дефолту isValid=false

sobbaka committed:
fix: удалил лишнюю проверку валидности

sobbaka committed:
add: форма невалидна если неверный формат фотографии

rakleed committed:
fix(front): восстановить выбор роли при регистрации

sobbaka committed:
add: валидация на ввод + валидация фото

sobbaka committed:
Merge branch 'develop' into ProfileCustomer_validation

sobbaka committed:
Merge pull request #275 from freelancing-platform-practicum/ProfileCustomer_validation

Profile customer validation

sobbaka committed:
add: добавил новую валидацию на форму регистрации

sobbaka committed:
Merge pull request #277 from freelancing-platform-practicum/signup_validation

add: добавил новую валидацию на форму регистрации

inigmat committed:
поле job_files можно не передавать при создании заказа

inigmat committed:
Merge pull request #278 from freelancing-platform-practicum/fix/back-jobfiles

поле job_files можно не передавать при создании заказа

alexandraspage committed:
валидация формы фрилансера

rakleed committed:
feat(front): добавить удаление заказа

MaximParamonov committed:
профиль пользователя создаётся при создании пользователя

MaximParamonov committed:
Merge branch 'bugfix/users' of github.com:freelancing-platform-practicum/freelancing-platform-project into bugfix/users

MaximParamonov committed:
Merge pull request #279 from freelancing-platform-practicum/bugfix/users

Bugfix/users

sobbaka committed:
add: валидация создания заказа

inigmat committed:
fix отображения заказов

inigmat committed:
Merge pull request #280 from freelancing-platform-practicum/fix/back-jobs

fix отображения заказов

inigmat committed:
fix отображения заказов

inigmat committed:
Merge pull request #281 from freelancing-platform-practicum/fix/back-jobs

fix отображения заказов

inigmat committed:
fix отображения заказов

inigmat committed:
Merge pull request #282 from freelancing-platform-practicum/fix/back-jobs

fix отображения заказов

sobbaka committed:
refactor: закомментил лишнее

inigmat committed:
fix ошибки валидации бюджета

inigmat committed:
Merge pull request #283 from freelancing-platform-practicum/fix/back-jobbudget

fix ошибки валидации бюджета

sobbaka committed:
add: поправил handleTaskSubmit, удалил комменты. добавил валидацию даты

sobbaka committed:
Merge branch 'develop' into task_validation

sobbaka committed:
Merge pull request #285 from freelancing-platform-practicum/task_validation

Валидация создания заказов

sobbaka committed:
save commit

sobbaka committed:
fix InputDocument

sobbaka committed:
Merge pull request #286 from freelancing-platform-practicum/fix_InputDocument

fix InputDocument

sobbaka committed:
fix: исправил ошибки в файле

sobbaka committed:
Merge pull request #287 from freelancing-platform-practicum/hotfix_useFormAndValidation

fix: исправил ошибки в файле

sobbaka committed:
Revert "fix: исправил ошибки в файле"

sobbaka committed:
Merge pull request #288 from freelancing-platform-practicum/revert-287-hotfix_useFormAndValidation

Revert "fix: исправил ошибки в файле"

sobbaka committed:
Revert "fix InputDocument"

sobbaka committed:
Merge pull request #289 from freelancing-platform-practicum/revert-286-fix_InputDocument

Revert "fix InputDocument"

sobbaka committed:
Revert "Валидация создания заказов"

sobbaka committed:
Merge pull request #290 from freelancing-platform-practicum/revert-285-task_validation

Revert "Валидация создания заказов"

sobbaka committed:
Merge branch task_validation into task_validation

sobbaka committed:
add: поправил тексты

sobbaka committed:
коммит перед мерджем с develop

sobbaka committed:
Merge branch 'develop' into task_validation

sobbaka committed:
поправил validateValues

sobbaka committed:
поправил handleTaskSubmit

sobbaka committed:
поправил ошибки линтера

sobbaka committed:
InputTags создает ошибки в общем errors

sobbaka committed:
Merge pull request #292 from freelancing-platform-practicum/task_validation

Валидация создания заказа + подружить с веткой develop

sobbaka committed:
change props in InputTags

sobbaka committed:
add: добавил фильтрация на главной

sobbaka committed:
Merge pull request #294 from freelancing-platform-practicum/main_filter

Добавил фильтр на главную страницу

sobbaka committed:
add: добавил фильтр фрилансеров по бюджету

inigmat committed:
исправил разрешение для редактирования заданий

sobbaka committed:
add: добавил поиск по заказам

sobbaka committed:
Merge pull request #297 from freelancing-platform-practicum/search_main

Добавлен поиск по заказам

sobbaka committed:
fix: поправил ошибку при сбросе фильтров

sobbaka committed:
Merge pull request #298 from freelancing-platform-practicum/search_main

fix: поправил ошибку при сбросе фильтров

inigmat committed:
Merge pull request #296 from freelancing-platform-practicum/fix/back-job-update

inigmat committed:
fix проблемы с БД и удаление ссылок на zohoverify

inigmat committed:
Merge pull request #299 from freelancing-platform-practicum/fix/back-job-update

fix проблемы с БД и удаление ссылок на zohoverify

inigmat committed:
fix редактирование заданий

inigmat committed:
Merge pull request #300 from freelancing-platform-practicum/fix/back-job-update

fix редактирование заданий

inigmat committed:
исправил частичное редактирование задания и добавил поиск по стэку

inigmat committed:
Merge pull request #301 from freelancing-platform-practicum/fix/back-job-update

исправил частичное редактирование задания и добавил поиск по стэку

inigmat committed:
исправил  поиск по стэку

inigmat committed:
Merge pull request #302 from freelancing-platform-practicum/fix/back-job-update

исправил  поиск по стэку

MaximParamonov committed:
Добавилен поиск по имени, фамилии, навыкам и специализации фрилансера, устранена (?) ошибка 500

MaximParamonov committed:
Merge pull request #303 from freelancing-platform-practicum/bugfix/users

Добавлен поиск по имени, фамилии, навыкам и специализации фрилансера…

sobbaka committed:
поменял метод в createUserProfile

sobbaka committed:
Поправил валидацию пароля

sobbaka committed:
Поправил отправку предложений в случае Жду предложений

sobbaka committed:
Merge pull request #304 from freelancing-platform-practicum/small_fixes

Небольшие правки

rakleed committed:
feat(front): добавить редактирование заказа

rakleed committed:
fix(front): восстановить бэкенд и убрать лишний отступ снизу у `InputText`

sobbaka committed:
add: сообщение, что поиск не дал результатов

sobbaka committed:
add: удаление лишних пробелов

sobbaka committed:
add: делать freelancers пустым если нет результатов

sobbaka committed:
поправл внешний вид главной

sobbaka committed:
в бюджет можно вводить только цифры + отображение ярлыков ОТ и ДО

alexandraspage committed:
нанять фрилансера и создать чат

sobbaka committed:
add: сделал кнопки предыдущая, следующая страница, вывод данных на клик по кнопкам

MaximParamonov committed:
фото, картинки дипломов и подобное выдаются в формате Base64, а не в виде ссылок на файлы

MaximParamonov committed:
Merge pull request #306 from freelancing-platform-practicum/bugfix/users

фото, картинки дипломов и подобное выдаются в формате Base64, а не в …

sobbaka committed:
добавил запрос задач фрилансера

sobbaka committed:
удалил console.log

sobbaka committed:
Merge branch 'develop' into main_page_development

sobbaka committed:
Merge pull request #308 from freelancing-platform-practicum/main_page_development

 Навигация + доработки мелких багов в поиске, фильтре

sobbaka committed:
fix API bug

sobbaka committed:
Merge pull request #309 from freelancing-platform-practicum/main_page_development

fix API bug
